server/session.py

```diff
@@ -269,9 +269,13 @@ def server_features(self):
     def is_tor(self):
         '''Try to detect if the connection is to a tor hidden service we are
         running.'''
-        tor_proxy = self.controller.peer_mgr.tor_proxy
-        peer_info = self.peer_info()
-        return peer_info and peer_info[0] == tor_proxy.ip_addr
+        try:
+            tor_proxy = self.controller.peer_mgr.tor_proxy
+            peer_info = self.peer_info()
+            return peer_info and peer_info[0] == tor_proxy.ip_addr
+        except Exception as e:
+            self.log_error('is_tor: {}'.format(e))
+            return False
 
```

Comment:
I agree. In fact, I made that a different commit with the intention of not including it in the PR. But my git skills are lacking and my PR ended up with both commits. My bad. I have more examples, if you need them. It only happens rarely and not at all since I made this change.